tests/components/dyson/common.py

```diff
@@ -49,7 +66,9 @@ async def async_update_device(
     hass: HomeAssistant, device: DysonDevice, state_type: Optional[Type] = None
 ) -> None:
     """Update the device using callback function."""
-    callback = device.add_message_listener.call_args[0][0]
+    callbacks = [args[0][0] for args in device.add_message_listener.call_args_list]
     message = MagicMock(spec=state_type)
-    await hass.async_add_executor_job(callback, message)
+    await asyncio.gather(
```

Comment:
This is ok since this is a test, but we wouldn't do this in production since it would claim threads greedily from the thread pool.

Reply:
So the point is, we should avoid using `asyncio.gather` with executor job, right? Then it is better to just do them one by one in a for loop.

Reply:
Yes, but if more than one sync call is needed it's best to create a function that does all sync calls and just schedule that function once on the executor.